src/Lanceur/Views/MainViewModel.cs

```diff
@@ -104,23 +105,36 @@ public class MainViewModel : ReactiveObject
                 .BindTo(this, vm => vm.Query);
 
             this.WhenAnyValue(vm => vm.Query)
-                .Throttle(TimeSpan.FromMilliseconds(100))
-                .Where(x => !x.IsNullOrWhiteSpace())
+                .Throttle(TimeSpan.FromMilliseconds(100), _schedulers.TaskpoolScheduler)
                 .Select(x => x.Trim())
-                .InvokeCommand(SearchAlias);
+                .Where(x => !x.IsNullOrWhiteSpace())
+                .Log(this, "Query changed.", x => $"{x}")
+                .ObserveOn(_schedulers.MainThreadScheduler)
+                .InvokeCommand(this, vm => vm.SearchAlias);
 
             this.WhenAnyValue(vm => vm.Results)
                 .Where(x => x is not null)
                 .Select(x => x.FirstOrDefault())
+                .Log(this, "Current alias.", x => $"'{(x?.Name ?? "<NULL>")}'")
+                .ObserveOn(_schedulers.MainThreadScheduler)
                 .BindTo(this, vm => vm.CurrentAlias);
 
-            var obs = this.WhenAnyObservable(vm => vm.SearchAlias, vm => vm.ExecuteAlias);
+            var obs = this
+                .WhenAnyObservable(vm => vm.SearchAlias, vm => vm.ExecuteAlias)
+                .ObserveOn(_schedulers.MainThreadScheduler);
+
             obs.Select(r => r.Results.ToObservableCollection())
+                .Log(this, "New results.", x => $"{x?.Count ?? -1} element(s)")
                 .BindTo(this, vm => vm.Results);
 
             obs.Select(r => r.KeepAlive)
                 .BindTo(this, vm => vm.KeepAlive);
 
+            this.WhenAnyObservable(vm => vm.SearchAlias, vm => vm.ExecuteAlias)
```

Comment:
Use `obs` instead of `this.WhenAnyObservable(vm => vm.SearchAlias, vm => vm.ExecuteAlias)`